src/nrnpython/nrnpy_nrn.cpp

```diff
@@ -2050,90 +2038,95 @@ static PyObject* section_getattro_safe(NPySecObj* self, PyObject* pyname) {
 static int section_setattro(NPySecObj* self, PyObject* pyname, PyObject* value) {
     Section* sec = self->sec_;
     if (!sec->prop) {
-        PyErr_SetString(PyExc_ReferenceError, "can't access a deleted section");
+        nrnpy_sec_referr();
         return -1;
     }
-    PyObject* rv;
-    int err = 0;
-    Py_INCREF(pyname);
+    auto _pyname_tracker = nb::borrow(pyname);  // keep refcount+1 during use
```

Comment:
This is a good, mechanical change, that's simple to review in isolation. To know that it's been changed correctly, I need to see all locations where an `Py_{INC,DEC}REF` occurs. Unfortunately, there's a lot of other changes that make it very hard for me to see if this has been done correct.

I'd like to suggest splitting the changes into:
1. Ref-counting changes.
2. Other rewrites that are possible now that we use RAII to do ref-counting.

Reply:
I hope I partially addressed this concern. Lmk if you find reviewable now